src/core/ui/components/sign-in/header.tsx

```diff
@@ -0,0 +1,54 @@
+import styles from './header.module.css';
+import { useAppearance } from '../../context/appearance';
+import React from 'react';
+
+const Wrapper = ({ children }) => (
+  <header className={styles.wrapper}>{children}</header>
+);
+
+const Title = <C extends React.ElementType>({
+  as,
+  children,
+}: {
+  as?: C;
```

Comment:
Can we rename this variable name? `as` is a keyword in certain languages and it is confusing to me.

Reply:
Yeah of course, what about `variant` or `component`? I used `as` since it is widely used in react component libraries.

Reply:
I think component is good!